x-pack/docs/en/security/securing-communications/security-basic-setup-https.asciidoc

```diff
@@ -0,0 +1,595 @@
+[[security-basic-setup-https]]
+=== Set up basic security for the Elastic Stack plus secured HTTPS traffic
+++++
+<titleabbrev>Set up basic security plus HTTPS</titleabbrev>
+++++
+
+In a production environment, some {es} features such as tokens and
+API keys will be disabled unless you enable TLS on the HTTP layer. This
+addition layer of security ensures that all communications to and from your
+cluster are secured.
+
+When you run the `elasticsearch-certutil` tool in `http` mode, the tool asks
+several questions about how you want to generate certificates. While there are
+numerous options, the following choices result in certificates that should
+work for most environments.
+
+[[signing-certificates]]
+.Signing certificates
+****
+The first question that the `elasticsearch-certutil` tool prompts you with is
+whether you want to generate a Certificate Signing Request (CSR). Answer
+`n` if you want to sign your own certificates, or `y` if you want to sign
+certificates with a central CA.
+
+[discrete]
+===== Sign your own certificates
+
+If you want to use the CA that you created when
+<<generate-certificates,Generating the certificate authority>> answer `n` to
+the first question. You then specify the location of your CA, which the tool
+uses to sign and generate a `.p12` certificate. The steps in this procedure
+follow this workflow.
+
+[discrete]
+===== Sign certificates with a central CA
+
+If you work in an environment with a central security team, they can likely
+generate a certificate for you. Infrastructure within your organization
+might already be configured to trust an existing CA, so it may be easier
+for clients to connect to {es} if you use a CSR and send that
+request to the team that controls your CA. To use a central CA, answer `y` to
+the first question.
+****
+
+[[basic-setup-https-prerequisites]]
+==== Prerequisites
+
+Complete all steps in <<security-basic-setup,Set up basic security for the Elastic Stack>>.
+
+[[encrypt-http-communication]]
+==== Encrypt HTTP client communications for {es}
+
+. Stop {es} and {kib} if they are running.
+
+. From the directory where you installed {es}, run the {es}
+   HTTP certificate tool to generate a Certificate Signing Request (CSR).
++
+[source,shell]
+----
+./bin/elasticsearch-certutil http
+----
++
+This command generates a `.zip` file that contains certificates and keys
+to use with {es} and {kib}. Each folder contains a `README.txt`
+explaining how to use these files.
+
+   a. When asked if you want to generate a CSR, enter `n`.
+
+   b. When asked if you want to use an existing CA, enter `y`.
+
+   c. Enter the path to your CA. This is the absolute path to
+   the `elastic-stack-ca.p12` file that you generated for your cluster.
```

Comment:
This can work. But I think we recommend to not share the CA between the transport and the http layers. The CA for transport layer should not be used for anything else since TLS is the only protection the inter-node communication has.